src/components/earlyadopter/ConnectAccount.vue

```diff
@@ -0,0 +1,221 @@
+                    {{ formatASTR(prevLockdropInfo.bonusAmt) }} ASTR
+                  </div>
+                  <div v-else>You don't deserve a bonus</div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, inject, ref, watch } from 'vue';
+import { keyring } from '@polkadot/ui-keyring';
+import allJson from '@/static/allContributors.json';
+import BN from 'bn.js';
+import MetamaskOption from '@/components/earlyadopter/MetamaskOption.vue';
+import AccountOption from '@/components/earlyadopter/AccountOption.vue';
+import { saveForBonusUser } from '@/db';
+import { UNIT, REWARD_RATIO } from '@/config/crowdloan';
+
+export default defineComponent({
+  components: {
+    MetamaskOption,
+    AccountOption
+  },
+  setup(props, { emit }) {
+    const api: any = inject('api');
+    const showConnectMetamsk = ref<boolean>(false);
+    const allAccounts = ref<string[]>([]);
+    const allAccountNames = ref<string[]>([]);
+    const allContributed = ref<BN[]>([]);
+    const allContributedD = ref<number[]>([]);
+    const isResultReward = ref<boolean>(false);
+    const prevLockdropInfo = ref();
+    const selAccount = ref(0);
+
+    const findContributed = async (addr) => {
+      let contributed = new BN(0);
+      const list: any = allJson;
+      for (const elem of list) {
+        if (elem.who === addr) {
+          contributed = new BN(elem.contributed);
+          // contributed = elem.contributed / 10 ** UNIT;
+          return contributed;
+        }
+      }
+      return contributed;
+    };
+
+    // set accounts from extensions
+    keyring.accounts.subject.subscribe((accounts: any) => {
+      if (api && accounts) {
+        // console.log('accounts', Object.keys(accounts));
+        allAccounts.value = Object.keys(accounts);
+        allAccountNames.value = Object.values(accounts).map((obj: any) =>
+          obj.option.name.replace('\n              ', '')
+        );
+      }
+    });
+
+    watch(
+      () => allAccounts.value,
+      async (accounts) => {
+        if (accounts) {
+          showConnectMetamsk.value = true;
+          allContributed.value = [];
+          for await (const account of allAccounts.value) {
+            const contributed = await findContributed(account);
+            // console.log('a', account + '/' + contributed.toString(10));
+            allContributed.value.push(contributed);
+            allContributedD.value.push(
+              contributed.div(new BN(10 ** UNIT)).toNumber()
+            );
+          }
+        }
+      }
+    );
+
+    const connectMetamask = async (ethAddr: string, ss58: string) => {
+      // ethAddr = '0x7d5aAD39da469B6496841215CeC5B14e3FcaDaDF';
+      console.log(ethAddr + '/' + ss58);
+      let response = await fetch('static/first-crowdloan.json');
+      const firstCrowdloan = await response.json();
```

Comment:
The name `firstCrowdloan` might be confusing here because we are talking about `lockdrop` events